### DISCUSSION TITLE
how to read WebRTC stream in Flutter?

### DISCUSSION THREAD

**Discussion by hiren3897** [bluenviron/mediamtx#1472] [Questions]
hello team,

Thanks for this fantastic work in the WebRTC stream provider

Can you give me  an example or POC on how I can read the WebRTC stream "wss://webrtc-demo..net:8889/StreamName/ws" on flutter apps.

Thanks in advanced

**Comment by aler9** [comment]
Hello, in order to integrate a WebRTC stream into a Flutter app, you can use one of the following techniques:

* integrate an iframe into the application, as described here: https://github.com/aler9/mediamtx#embedding-1
* copy and paste the JavaScript class contained here https://github.com/aler9/mediamtx/blob/main/internal/core/webrtc_index.html into the app, in order to allow the app to initialize a WebRTC stream from a WebSocket URL.
* wait for this feature to be developed: https://github.com/aler9/mediamtx/issues/1429 WHIP is a standard format that allows to initialize WebRTC streams from URLs, therefore it should be enough to pass a WHIP URL to a compatible library.

**Reply by pikachu937** [reply]
Hello. I join the author, we really need wss in applications for android / ios mobile devices.

[a comment by an automated account is not shown]
